### PR TITLE
feat(useGiftVoucherFilter): set gift voucher unavailable in store api

### PR DESCRIPTION
**Pull request by michnhokn**
We have decided to make the `GIFT_VOUCHER` product unavailable via the proxy.
This PR contains a new function to identify a product entity in the response from the original API and check for the gift voucher product id. If found, some properties will be overridden to make the product appear unavailable in the application.

### CODE REVIEW

File: utils/useGiftVoucherFilter.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// We need to prevent the product GIFT_VOUCHER from being available via the StoreAPI right now.
+export function useGiftVoucherFilter(item: unknown) {
+    if (
+        item?.apiAlias === 'product' &&
+        item?.id === '3b6878fb7b9b4504a30d07ffe851cca3'
```
Comment (marcojetson) [inline utils/useGiftVoucherFilter.ts]: can we check SKU instead?

File: utils/usePrepareRequest.ts [hunk 2]
```diff
@@ -5,34 +5,18 @@ export async function usePrepareRequest(event: H3Event) {
...
-        requestHeaders['sw-include-seo-urls'] = headers['sw-include-seo-urls'];
-    }
+    const passHeaders = [
+        'sw-language-id',
+        'sw-access-key',
+        'sw-context-token',
+        'sw-include-seo-urls',
+    ];
+
+    passHeaders.forEach((passHeader) => {
```
Comment (marcojetson) [inline utils/usePrepareRequest.ts:15]: this was not really needed :P

Comment (michnhokn) [reply utils/usePrepareRequest.ts:15]: Nah but it's more readable.

File: utils/useGiftVoucherFilter.ts (hunk 1 above)
Comment (michnhokn) [reply utils/useGiftVoucherFilter.ts]: sure thing.

**Pull request merged**